CommonTools/PileupAlgos/test/recomputePuppiOnMiniAOD.py

```diff
@@ -0,0 +1,142 @@
+# Auto generated configuration file
+# using: 
+# Revision: 1.19 
+# Source: /local/reps/CMSSW/CMSSW/Configuration/Applications/python/ConfigBuilder.py,v 
+# with command line options: step5 --conditions auto:run2_mc --nThreads 2 -n 10 --era Run2_2016 --eventcontent MINIAODSIM --filein file:step3_inMINIAODSIM.root -s NANO --datatier NANOAODSIM --mc --fileout file:step5.root
+import FWCore.ParameterSet.Config as cms
+
+from Configuration.Eras.Era_Run2_2016_cff import Run2_2016
+
+process = cms.Process('NANO',Run2_2016)
+
+# import of standard configurations
+process.load('Configuration.StandardSequences.Services_cff')
+process.load('SimGeneral.HepPDTESSource.pythiapdt_cfi')
+process.load('FWCore.MessageService.MessageLogger_cfi')
+process.load('Configuration.EventContent.EventContent_cff')
+process.load('SimGeneral.MixingModule.mixNoPU_cfi')
+process.load('Configuration.StandardSequences.GeometryRecoDB_cff')
+process.load('Configuration.StandardSequences.MagneticField_cff')
+process.load('PhysicsTools.NanoAOD.nano_cff')
```

Comment:
what do you take from nano_cff ?
the recomputePuppiOnMiniAOD.py is defacto a nano step:
here you seems to cut and past what is done in nano
best would be to isolate the puppi recomputation in a separate function and then use this both in nano_cff and in this test code

Reply:
indeed there is no usage of the nano_cff here and I've removed this line.
I agree having a dedicated function taking care of recomputing all MiniAOD content (including AK4, AK8, MET) based on PUPPI from MiniAOD (i.e. Mini-from-Mini) would be very useful, such that Nano would only need to turn a switch to do this. Let's do this in a dedicated PR, it is quite involved as a new produced is likely needed to make this possible for AK8 jets.
The test config here is based on what's possible so far, to have a way to test this PR.